src/measure_theory/integral/average.lean

```diff
@@ -384,5 +537,157 @@ by simpa only [average_eq_integral]
   using exists_not_mem_null_average_le (is_probability_measure.ne_zero μ) hf hN
 
 end probability_measure
-end first_moment
+end first_moment_real
+
+section first_moment_ennreal
+variables {N : set α} {f : α → ℝ≥0∞}
+
+/-- **First moment method**. A measurable function is smaller than its mean on a set of positive
+measure. -/
+lemma measure_le_set_laverage_pos (hμ : μ s ≠ 0) (hμ₁ : μ s ≠ ∞)
+  (hf : ae_measurable f (μ.restrict s)) (hs : null_measurable_set s μ) :
+  0 < μ {x ∈ s | f x ≤ ⨍⁻ a in s, f a ∂μ} :=
+begin
+  obtain h | h := eq_or_ne (∫⁻ a in s, f a ∂μ) ∞,
+  { simpa [mul_top, hμ₁, laverage, h, top_div_of_ne_top hμ₁, pos_iff_ne_zero] using hμ },
+  have := measure_le_set_average_pos hμ hμ₁ (integrable_to_real_of_lintegral_ne_top hf h),
+  rw [←set_of_inter_eq_sep, ←measure.restrict_apply₀' hs],
+  rw [←set_of_inter_eq_sep, ←measure.restrict_apply₀' hs,
+    ←measure_diff_null (measure_eq_top_of_lintegral_ne_top hf h)] at this,
+  refine this.trans_le (measure_mono _),
+  rintro x ⟨hfx, hx⟩,
+  dsimp at hfx,
+  rwa [←to_real_laverage hf, to_real_le_to_real hx (set_laverage_ne_top hμ h)] at hfx,
+  simp_rw [ae_iff, not_ne_iff],
+  exact measure_eq_top_of_lintegral_ne_top hf h,
+end
+
+/-- **First moment method**. A measurable function is greater than its mean on a set of positive
+measure. -/
+lemma measure_set_laverage_le_pos (hμ : μ s ≠ 0) (hf : ae_measurable f (μ.restrict s))
```

Comment:
you don't need both hf and hs: the result is true assuming either one or the other, if I'm not mistaken. The more useful version would be assuming only hs, because checking that a function is measurable is often more work than checking that a set is measurable. Hint for the proof: use `exists_measurable_le_lintegral_eq.`